testsuite/config/tools.py

```
@@ -52,12 +52,14 @@ def _fetcher(settings, _):
     return _fetcher
 
 
-def fetch_service_ip(name, port: int, protocol: str = "http"):
+def fetch_service_ip(name, port: int, protocol: str = "http", namespace: str = None):
```

Ruff (0.15.12)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)